testsuite/cluster-test/src/report.rs

```diff
@@ -43,6 +43,14 @@ impl SuiteReport {
         self.text.push_str(&text);
     }
 
+    pub fn remove_newline(&mut self) {
+        if !self.text.is_empty() {
+            if self.text.ends_with('\n') {
+                self.text.pop();
+            }
+        }
```

Comment:
```suggestion
        
            if self.text.ends_with('\n') {
                self.text.pop();
            }
   
```

You don't need to check if its empty before checking if it ends with a newline